### PR TITLE
test(claude-code): comprehensive build output test suite

### PR DESCRIPTION
**Pull request by MacAttak**
## Summary

- Add 155-assertion test suite (`tests/test-claude-code-build.sh`) verifying all aspects of the claude-code adapter build output
- Add `test-claude-code` CI job to `validate.yml` for PR validation
- Brings claude-code adapter to test parity with the opencode adapter (audit finding F6)

## Acceptance Criteria

| # | Criterion | Status |
|---|-----------|--------|
| AC-1 | Test file with helpers (pass, fail, assert_eq, extract_frontmatter, extract_body) | PASS |
| AC-2 | Build output structure (skills/19, agents/6, protocols/19, hooks, .claude-plugin, CLAUDE.md, README.md) | PASS |
| AC-3 | Identity mapping — tools stay uppercase across ALL skills | PASS |
| AC-4 | Agent identity — no mode:subagent, shorthand models, array-style tools | PASS |
| AC-5 | Protocol refs — bare `protocols/` paths, no `.specwright/protocols/` | PASS |
| AC-6 | Hook files valid JavaScript (`node --check`) + match adapter source | PASS |
| AC-7 | Plugin manifest valid JSON with name and version fields | PASS |
| AC-8 | No platform markers remain in ANY output file | PASS |
| AC-9 | sw-build Claude Code content preserved (Task tools, Task tracking, Mid-build checks) | PASS |
| AC-10 | No opencode-specific content (no commands/, package.json, plugin.ts) | PASS |
| AC-11 | Source files not modified by build | PASS |
| AC-12 | Test added to CI pipeline (`test-claude-code` job in validate.yml) | PASS |
| AC-13 | Test exits 0 with 0 failures | PASS |

## Gate Results

| Gate | Status | Findings (B/W/I) |
|------|--------|-------------------|
| gate-build | SKIP | 0/0/0 — no build/test commands configured |
| gate-tests | WARN | 0/1/0 — hardcoded counts (accepted project pattern) |
| gate-security | PASS | 0/0/0 |
| gate-wiring | PASS | 0/0/0 |
| gate-spec | PASS | 0/0/0 — 13/13 criteria mapped |

## Evidence

- `build-report.md` — SKIP (no commands configured)
- `test-quality.md` — WARN (hardcoded counts match opencode test pattern)
- `security-report.md` — PASS (no secrets, no injection, safe CI config)
- `wiring-report.md` — PASS (test integrated in CI, no orphans)
- `spec-compliance.md` — PASS (13/13 criteria with file:line evidence)

## Work Unit

`claude-code-tests` — unit 2 of 3 in `audit-remediation`

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: tests/test-claude-code-build.sh [hunk 1]
```diff
@@ -0,0 +1,906 @@
...
+  local closing_line
+  closing_line=$(tail -n +2 "$file" | grep -n '^---$' | head -n 1 | cut -d: -f1)
+  if [ -z "$closing_line" ] || [ "$closing_line" -lt 1 ]; then
+    return 1
+  fi
+  tail -n +"$((closing_line + 2))" "$file"
+}
+
+# Extract allowed-tools list items from YAML frontmatter
+extract_allowed_tools() {
```
[the remark this answers is not in the document]
Comment (MacAttak) [reply tests/test-claude-code-build.sh]: Fixed in dcda8c9. Changed stop pattern from `^[a-z]` to `^[^ ]` which matches any non-indented line — correctly handles both lowercase and uppercase YAML keys, and the final-field edge case.

File: tests/test-claude-code-build.sh [hunk 2]
```diff
@@ -0,0 +1,906 @@
...
+echo "--- Core skill content integrity ---"
+
+# Verify a few core files have not been altered by checking git status
+for skill in sw-init sw-build sw-verify; do
+  core_skill="$ROOT_DIR/core/skills/$skill/SKILL.md"
+  if [ -f "$core_skill" ]; then
+    # Check the file still starts with frontmatter
+    FIRST_LINE=$(head -n 1 "$core_skill")
+    assert_eq "$FIRST_LINE" "---" "core/$skill/SKILL.md still starts with ---"
+  fi
```
[the remark this answers is not in the document]
Comment (MacAttak) [reply tests/test-claude-code-build.sh]: Fixed in dcda8c9. Replaced the three spot-check assertions with a single `git diff --exit-code -- core/ adapters/` check that catches any byte-level modification to any source file. Much stronger coverage.
